src/util/url-filter.js

```diff
@@ -0,0 +1,23 @@
+//Remove unnecessary query params from the URL to avoid redundancy
+const NORMALIZE_OPTS = {
+    'stripWWW': false,
+    'removeDirectoryIndex': false,
+    'removeTrailingSlash': false,
+    'stripFragment': true,
+    'normalizeProtocol': false
+}
+const GSEARCH_SUBSTR = 'www.google'
+const FB_SUBSTR = 'www.facebook'
+const normalizeUrl = require("normalize-url")
+export default function urlFilter(url) {
+    const parts = url.split('.')
+
+    if (parts[0] == 'www' && parts[1] == 'google') {
+        // Case 1: Google Search
+        return normalizeUrl(url.split('&')[0], NORMALIZE_OPTS)
```

Comment:
These `String.prototype.split` calls don't really belong here either. The purpose of these parts would be to run `normaliseUrl` for a second time on the already normalized URL, passing in rules that will ensure that certain query params get removed (and all other normalization features disabled).

Ideally if these rules were in a structure that maps the domains to rules, there wouldn't be any need for conditional branches:

```js
const domain = ... // some logic to get the domain (or whatever uniquely identifies this page's rules)

const undesirableParams = dictOfRules.get(domain)

return undesirableParams
  ? normalizeUrl(url, { removeQueryParameters: undesirableParams })
  : url // There was no entry in the dictionary under this domain, so return the input URL
```

Reply:
The rules as Oliver mentioned are pretty abstract so I can't put them in a dictionary. 
> *For google searches, we need to remove everything after the first &,
*For Facebook we can remove all querystrings, only for the imports we need to remove ALL google maps links, so everything that has
*ALL hashes need to be removed for EVERY url.

So for now, I'll be putting `if...else` clauses only for now. Earlier on I had imagined it to be the way you've written above.

Reply:
I wrote about why these rules were not a good idea [here](https://github.com/WorldBrain/WebMemex/issues/112#issuecomment-340231843). And we later settled on the 3 rules:

> So for google, facebook and twitter it would then be the standard search querystring.

"standard search querystring" as in the parameter `q`.

Regardless of how the rules are, they need to be stored in some kind of data structure which the code performs a lookup using. We can't do much with it if the rules are all embedded within a series of conditional branches.